### DISCUSSION TITLE
Does support model hot update

### DISCUSSION THREAD

**Discussion by MrRace** [bentoml/BentoML#1204] [Q&A] [answered]
If I want to  update my model and keep my service alive（hot update）, Could  the BentoML support it? 
Thanks a lot!

**Comment by parano** [comment, marked as answer]
Hi @MrRace 

Model hot update is not supported in BentoML and currently not a priority for us. We do plan to build some base APIs for advanced users to implement dynamic model loading/unloading strategy themselves, but it won't be the recommended way of doing things in BentoML.

For most users, we'd recommend using existing blue/green deployment tools to achieve zero downtime update of new model. So that DevOps can treat the ML model serving workloads the same way how other micro-services get deployed.

**Comment by MrRace** [comment]
@parano Could you recommend some blue/green deployment tools ? Thanks！

**Reply by yubozhao** [reply]
@MrRace 
If your production is using kubernetes, I recommend to check out istio or similar services.  

For istio, you can find blue/green deployment under the section traffic shifting: https://istio.io/latest/docs/tasks/traffic-management/traffic-shifting/

**Reply by MrRace** [reply]
For  single local machine, any advise ?

**Reply by parano** [reply]
@MrRace try docker's own rolling-update? https://docs.docker.com/engine/swarm/swarm-tutorial/rolling-update/
